### PR TITLE
Feature/309 refactor into async

### PR DESCRIPTION
**Pull request by tracetoio-weilong**

### CODE REVIEW

File: flow/helper.js [hunk 1]
```diff
@@ -21,172 +21,124 @@ exports.initialize = function(outerCallback){
...
-
-exports.checkBalance = function(outerCallback){
+  let gasPrice = await this.alice.getGasPrice();
+  this.alice.updateGasPrice(gasPrice);
+  this.bob.updateGasPrice(gasPrice);
+  this.carol.updateGasPrice(gasPrice);
+  this.dave.updateGasPrice(gasPrice);
+  this.eve.updateGasPrice(gasPrice);
+  console.log('Initialize finished.');
+
```
Comment (tracetoio-dias) [inline flow/helper.js]: Extra space

File: models/ToBeServiceProvider.js [hunk 2]
```diff
@@ -7,11 +7,19 @@ class ToBeServiceProvider extends TTEntity {
...
+  apply(ratePerService, reputation, companyName, email, uriForRubrics, hashFroRubrics){
+    const _this = this;
+    return new Promise(function(resolve, reject) {
+      _this.w3.sendToContractbyIdx(_this.SPListContract, 'addPendingSP', _this.gasPrice*4, _this.w3.getWalletAddress(), ratePerService, reputation, companyName, email, uriForRubrics, hashFroRubrics)
+        .then(data => resolve(data), reason => reject(reason));
+    });
   }
   getStatus(callback){
-    return this.w3.callContractbyIdx(this.SPListContract, 'isSP', callback, this.w3.getWalletAddress());
+    const _this = this;
```
Comment (tracetoio-dias) [inline models/ToBeServiceProvider.js]: can we check on how we can avoid this pattern

File: models/Verifier.js [hunk 3]
```diff
@@ -12,30 +12,52 @@ class Verifier extends TTEntity {
...
+      const unlockProfileContract = httpw3.addContract('UnlockProfile', environment.TraceToUnlockProfile.address, environment.TraceToUnlockProfile.abi);  
+      httpw3.sendToContractbyIdx(unlockProfileContract, 'setKey', _this.gasPrice*4, profileHash, keyPiece, requestor)
+      .then(data => resolve(data), reason => reject(reason));
+    });
   }
 
-  getProfileCount(user, callback){
-    return this.w3.callContractbyIdx(this.ProfileContract, 'getUserProfileTokenCount', callback, user);
+  getProfileCount(user){
+    const _this = this;
```
Comment (tracetoio-dias) [inline models/Verifier.js]: Here do we use _this later?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "traceto-workflow-lib",
-  "version": "1.0.25",
```
Comment (tracetoio-dias) [inline package.json]: Shouldnt we bump up major version

File: flow/helper.js (hunk 1 above)
Comment (tracetoio-weilong) [reply flow/helper.js]: removed

File: models/Verifier.js (hunk 3 above)
Comment (tracetoio-weilong) [reply models/Verifier.js]: yay, updated for verifier

File: package.json (hunk 4 above)
Comment (tracetoio-weilong) [reply package.json]: upgraded to 2.0.0

File: models/ToBeServiceProvider.js (hunk 2 above)
Comment (tracetoio-weilong) [reply models/ToBeServiceProvider.js]: Will improve this later

**Pull request merged**